src/jit_channel/channel_manager.rs

```diff
@@ -192,13 +198,19 @@ enum OutboundJITChannelState {
 		payment_size_msat: Option<u64>,
 		opening_fee_params: OpeningFeeParams,
 	},
+	PendingPaymentReceived {
```

Comment:
Yeah good point.  I think even can/should get rid of the first state of InvoiceParametersGenerated because really soon as that is given out you are already awaiting payment with an empty vec of htlcs received.

Reply:
well the state is really supposed to be that we received _part_ of the payment.  it only transitions here once the first htlc is received.

i wonder though if we should just get rid of the InvoiceParametersGenerated state and just immediately enter some kind of AwaitingPayment state that holds the htlcs and initializes with empty vec.

Reply:
I ended up removing both states and now the initial state of an outbound channel is `AwaitingPayment`. If mpp is enabled it stays there until receives enough to cover payment_size+fee.  If mpp not enabled then after the first htlc is received, if it's not enough it fails immediately.